src/commands/domain.js

```diff
@@ -89,4 +92,206 @@ async function rm (params) {
   Logger.println('Your domain has been successfully removed');
 }
 
-module.exports = { list, add, getFavourite, setFavourite, unsetFavourite, rm };
+export async function diagApplication (params) {
+  let hasError = false;
+  const allDomainDiagnostics = [];
+  const { alias, app: appIdOrName, format, domain } = params.options;
+  const { ownerId, appId } = await Application.resolveId(appIdOrName, alias);
+  const hasDomainFilter = domain.length > 0;
+
+  const app = await getApp({ id: ownerId, appId }).then(sendToApi);
+  const expectedDnsForPublicLoadBalancer = await getDefaultLoadBalancersDnsInfo({ id: ownerId, appId }).then(sendToApi);
+
+  const loadBalancerDnsConfig = {
+    aRecords: expectedDnsForPublicLoadBalancer[0].dns.a,
+    cnameRecord: expectedDnsForPublicLoadBalancer[0].dns.cname,
+  };
+
+  const filteredDomains = app.vhosts.filter(({ fqdn }) => fqdn.includes(domain));
+  const domains = getParsedDomains(filteredDomains);
+  const sortedDomains = sortDomains(domains);
+
+  for (const { hostname, pathPrefix, isApex } of sortedDomains) {
+    const resolvedDnsConfig = {
+      aRecords: await resolver.resolveA(hostname),
+      cnameRecords: await resolver.resolveCname(hostname) ?? [],
+    };
+
+    const diagnosticResults = diagDomainConfig({ hostname, pathPrefix, isApex }, resolvedDnsConfig, loadBalancerDnsConfig);
+    allDomainDiagnostics.push({ ...diagnosticResults, resolvedDnsConfig });
+
+    if (diagnosticResults.diagSummary === 'invalid' || diagnosticResults.diagSummary === 'no-config') {
+      hasError = true;
+    }
+  }
+
+  switch (format) {
+    case 'json':
+      Logger.printJson(allDomainDiagnostics);
+      break;
+    default: {
+      if (allDomainDiagnostics.length > 0) {
+        Logger.println(colors.yellow('\n⚠ TODO avec David, une phrase pour dire qu\'on ne supporte pas tout + se référer à la doc de la commande + un lien vers feedback ?'));
```

Comment:
nitpick: do not forget this todo

Reply:
Discussed with David & Hubert, we'll remove this warning, these will be included in docs but not the command itself.